pkg/lexer/lexer.go

```diff
@@ -20,38 +20,46 @@ import (
 	"github.com/raklaptudirm/mash/pkg/token"
 )
 
+// Various error values returned by lexer.consume.
 var (
 	ErrIllegalNUL = fmt.Errorf("illegal character NUL")
 	ErrIllegalBOM = fmt.Errorf("illegal byte order mark")
 	ErrIllegalEnc = fmt.Errorf("illegal utf-8 encoding")
 )
 
+// lexer represents a mash source string and related lexing information.
 type lexer struct {
-	src string
-	ch  rune
+	src string // source string
+	ch  rune   // current character
 
 	insertSemi bool
 
-	Tokens chan token.Token
+	Tokens chan token.Token // lexer token channel
 
-	err ErrorHandler
+	err ErrorHandler // lexer errors handling function
 
-	offset   int
-	rdOffset int
+	offset   int // offset of the start of the token in the source
+	rdOffset int // offsen of the current rune in the source
```

Comment:
what is `rd` in this `rdOffset`?

Reply:
`rdOffset` stands for the "read offset", or the offset of the `rune` we are currently reading in the source.

Reply:
```suggestion
	rdOffset int // offset of the current rune in the source
```

Maybe just call it `curRune` or `curIndex` since this is only being used for looking up the current position of the rune in the source. Offset is more confusing, I thought you meant something completely different here.